include/etl/private/delegate_cpp11.h

```diff
@@ -434,8 +737,20 @@ namespace etl
     }
 
   private:
+    struct lambda_tag {};
+
+    union object_type {
+      void* pointer;
+      alignas(void*) char buffer[sizeof(void*)];
 
-    using stub_type = TReturn(*)(void* object, TParams...);
+      ETL_CONSTEXPR14 object_type() : pointer{nullptr} {} // None constructor
+      ETL_CONSTEXPR14 object_type(void* ptr) : pointer {ptr} {} // object pointer 
+      template <typename T>
+      ETL_CONSTEXPR14 object_type(lambda_tag, T&& object) : buffer{0, } { new (&this->buffer[0]) T(etl::forward<T>(object)); }
+
+      ~object_type() = default;
```

Comment:
Is this okay to default the destructor when you have `new`'d the buffer https://github.com/ETLCPP/etl/pull/867/files#diff-e37e1a7180ece428e4933802e623de4e0e17c620c429fc52653c2ad00487e5b5R749 here?  

IIRC you need to call the destructor for T after the placement new is used.

Reply:
Yeah, you're right. Forgot that one... but if we add `std::is_destructible` as a guard, then we should be safe, right?

Reply:
Using `std::is_trivially_destructible` would make it impossible for use with the `ETL_NO_STL` option. Implementing a cross platform `etl::is_trivially_destructible` is not simple as it relies on relevant compiler intrinsics being available.

Reply:
Also I would not want the destructor implemented. Currently we use delegates on our project and are able to store them in ROM.  Once it's not trivially destructible, that goes away which would ruin a lot of our data.  I think we would be left going back to the bad ol' days of straight function pointers :(

Reply:
I think that the default assignment makes it trivially destructible, or at least is what I thought when I was inspired by the following blog post: https://akrzemi1.wordpress.com/2012/12/13/constexpr-unions/